### DISCUSSION TITLE
Anyone having issue to run ethers in Snap?

### DISCUSSION THREAD

**Discussion by ohyung1287** [MetaMask/snaps#962] [Q&A] [answered]
I was trying to connect my smart contract in Snap so I followed the ethers-js example to initialize a provider instance like this
`const provider = new ethers.providers.Web3Provider(wallet);`
When I try the `await provider.getBlockNumber();` in Snap to see if this provider really works, it just never responds until timeout.
Does anyone have the same problem before or can you share how you connect ethers in Snap?
Thank you.

**Comment by ritave** [comment, marked as answer]
Hello @ohyung1287.

That should be working as intended. Does anything appear in your Console log of the DevTools?

Snap errors can also appear in the background.html log as well.
To access the background.html log, first [enter the list of extension](chrome://extensions/), enable developer mode in top right and then click on `background.html` page of the MetaMask Flask extension

**Reply by ohyung1287** [reply]
Thank you @ritave.  I did not know how to look console before you told me. 
The error is still The RPC request timed out when I try to call provider function like await provider.getBlockNumber or view smart contract data instance.name() for example. 
I'll try to investigate more and see why provider not working only in Snap.

**Reply by ohyung1287** [reply]
Found the issue, it was because my provider initialized in global while wallet obj is undefined atm. By moving initialization inside the onRpcRequest solve the issue. Thanks a lot for you help
